typescript/codepipeline-build-deploy/lib/codepipeline-build-deploy-stack.ts

```diff
@@ -54,10 +50,31 @@ export class CodepipelineBuildDeployStack extends cdk.Stack {
       portMappings: [{ containerPort: 80 }],
     });
 
+    const codeBuildServiceRole = new iam.Role(this, 'CodeBuildServiceRole', {
+      assumedBy: new iam.ServicePrincipal('codebuild.amazonaws.com'),
+      roleName: 'CodeBuildServiceRole',
+      description: 'Service Role for CodeBuild',
+      managedPolicies: [
+        iam.ManagedPolicy.fromAwsManagedPolicyName('CloudWatchLogsFullAccess'),
+        iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonS3ReadOnlyAccess'),
+        // Add any other necessary managed policies or can be fine grained as per usgae
+      ]
+    });
+
+    // Grant access to Secrets Manager
+    codeBuildServiceRole.addToPolicy(
+      new iam.PolicyStatement({
+        effect: iam.Effect.ALLOW,  
+        actions: ['secretsmanager:GetSecretValue'],
+        resources: ['arn:aws:secretsmanager:<aws_region>:<aws_account>>:secret:github-oauth-token-*'],
```

Comment:
This value needs to be updated. Please add documentation in the README.md or have another way to get this value.